### ISSUE TITLE
Increment level headers by 1

### ISSUE THREAD

**Issue by remcohaszing** [wooorm/remark-api#4]
I just gave this project a try for [`estree-util-value-to-estree`](https://github.com/remcohaszing/estree-util-value-to-estree) (not committed). Level 6 headers were generated for _Fields_, _Parameters_, and _Returns_, but level 3 headers for the export names. This triggers [`remark-lint-heading-increment`](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-heading-increment), which I feel is a fair rule. Perhaps all headers should only ever increment by 1.

**Comment by wooorm** [comment]
The behavior you see is intentional.

But see: https://github.com/wooorm/remark-api/issues/3

**Comment by remcohaszing** [comment]
Yes, I felt that’s somewhat related, but not the same. This is more intended as early feedback than a bug report. :)

**Comment by wooorm** [comment]
I agree that it’s not optimal to use h6. But I think h4 isn’t great either, it becomes too big, too important. And a sole strong in a paragraph is bad too.

**Comment by remcohaszing** [comment]
I do agree with your points. They’re all trade-offs.

A benefit of using `h4`, is that `h5`, and `h6` can be safely used inside the JSDoc description.

**Comment by wooorm** [comment]
Ahh, that’s a very interesting point, headings inside the JSDoc markdown!

Though, would people use `h5` there? Wouldn’t they start at `h1`/`h2`?
Perhaps we’d need to “fix” the heading structured based on the user’s JSDoc markdown and readme markdown?

**Comment by remcohaszing** [comment]
IMO, if you use this library, you can adjust your JSDoc a bit to work with it nicely.

However, header levels 5 and 6 render ridiculously small in VSCode hovers IMO.

![Header levels in VSCode hover info](https://github.com/wooorm/remark-api/assets/779047/bfb88fe9-7ad2-4e5c-b701-d3f740b1a962)

> Perhaps we’d need to “fix” the heading structured based on the user’s JSDoc markdown and readme markdown?

This would be great :100:

**Comment by wooorm** [comment]
>  if you use this library, you can adjust your JSDoc a bit to work with it nicely.

Agreed, but I’d *also* want for people to write docs to that also work good with other tools. So, the hover thing you mention. Things like that are good reasons to go one way or the other
